### PR TITLE
enhancement/reformating-slug

### PR DESCRIPTION
**Pull request by aminsaedi**

### CODE REVIEW

**Comment by aminsaedi** [comment]
it seems that gatsby convert file name to kebab case. so I changed everywhere we use link to another page (`toc` & `AppLink`) to use `kebab-case-slug`

**Comment by aminsaedi** [comment]
Also I added one line to `.gitignore` to prevent adding `.ds_store` file

File: src/components/appLink/appLink.tsx [hunk 1]
```diff
@@ -26,8 +27,9 @@ function AppLink({
 
   if (href.startsWith(":")) {    
     const slug = data.allMdx.nodes.find((i) => i.frontmatter?.id === href.substring(1))?.slug;
+    const formattedSlug = slug.split("/").map(i => paramCase(i)).join("/")
```
Comment (pmvald) [inline src/components/appLink/appLink.tsx:30]: This function is repeating in several places, you can move this logic into a util file, like `src/util/format-slugs.ts`

**Review by pmvald — CHANGES_REQUESTED** [review]
When renaming mdx files, we still have numerical prefixes, like:

03-sectionThree
  010-topicOne
  021-sectionTwo
  030-anyNameIwant

And we want slugs to become:

section-three/topic-one
section-three/section-two
section-three/any-name-i-want

Why we have those numbers in the beginning? Because I want my topics to have the same order on the file system as in the doc TOC. recall we sort the mdx paths by name. So without those names the last topic (anyNameIwant) will show up first in TOC, but it should be in the third place! 

This requires us to transform the slugs generated by Gatsby - might not be easy. We may need to include a plugin to transform slugs and instead of filename `{mdx.slug}.ts` we may need to use `{ourOwnNode.slug}` ... see if it's an easy task or needs much effort.

**Comment by aminsaedi** [comment]
I have not found any useful plugin yet...
still searching for a way to do it...
Is that the purpose of it is to remove (`03`-`02`-...) from the url (`http://localhost:8000/docs/pgweb/03-section/02-subsection/02-topic`) ?

**Comment by pmvald** [comment]
> I have not found any useful plugin yet... still searching for a way to do it... Is that the purpose of it is to remove (`03`-`02`-...) from the url (`http://localhost:8000/docs/pgweb/03-section/02-subsection/02-topic`) ?

Yes, we want to remove numbers from the url. Numbers are only used for sorting files in the folder in exact same order as they appear in the doc, but they should not be in the url itself.

I don't think you can find a plugin for this particular purpose. We need to write a *data transformer* to convert the node *mdx* so a custome node type *which is the same as mdx except the slugs have their numbers removed)

Youcan take a look at this guide: https://www.gatsbyjs.com/docs/how-to/plugins-and-themes/creating-a-transformer-plugin/

Similar to this guide, we can create a new node, say `processedMdx`, out of node `mdx` which is basically the same as this node except the slugs are modified. Then we use this node instead of mdx (i.e., rename the file `{mdx.slug}.ts` to `{processedMdx.slug}.ts`)

If you still couldn't do that then assign the task to me. For today we need to build some background on monorepo concept (as described in the message I sent today). So if this task became hard just leave it and focus on the monorepo study instead.

**Pull request closed without merging**